lib/redcord/actions.rb

```diff
@@ -26,37 +26,62 @@ module ClassMethods
 
     sig { params(args: T::Hash[Symbol, T.untyped]).returns(T.untyped) }
     def create!(args)
-      args[:created_at] = args[:updated_at] = Time.zone.now
-      instance = TypeCoerce[self].new.from(args)
-      id = redis.create_hash_returning_id(model_key, to_redis_hash(args))
-      instance.send(:id=, id)
-      instance
+      Redcord::Base.trace(
+       'redcord_actions_class_methods_create!',
+        model_name: name,
+      ) do
+        args[:created_at] = args[:updated_at] = Time.zone.now
+        instance = TypeCoerce[self].new.from(args)
+        id = redis.create_hash_returning_id(model_key, to_redis_hash(args))
+        instance.send(:id=, id)
+        instance
+      end
     end
 
     sig { params(id: T.untyped).returns(T.untyped) }
     def find(id)
-      instance_key = "#{model_key}:id:#{id}"
-      args = redis.hgetall(instance_key)
-      if args.empty?
-        raise Redcord::RecordNotFound, "Couldn't find #{name} with 'id'=#{id}"
+      Redcord::Base.trace(
+       'redcord_actions_class_methods_find',
+        model_name: name,
+      ) do
+        instance_key = "#{model_key}:id:#{id}"
+        args = redis.hgetall(instance_key)
+        if args.empty?
+          raise Redcord::RecordNotFound, "Couldn't find #{name} with 'id'=#{id}"
+        end
+
+        coerce_and_set_id(args, id)
       end
-
-      coerce_and_set_id(args, id)
     end
 
     sig { params(args: T::Hash[Symbol, T.untyped]).returns(T.untyped) }
     def find_by(args)
-      where(args).to_a.first
+      Redcord::Base.trace(
+       'redcord_actions_class_methods_find_by_args',
+        model_name: name,
+      ) do
+        where(args).to_a.first
+      end
     end
 
     sig { params(args: T::Hash[Symbol, T.untyped]).returns(Redcord::Relation) }
     def where(args)
-      Redcord::Relation.new(T.let(self, T.untyped)).where(args)
+      Redcord::Base.trace(
```

Comment:
Where doesn't actually make any Redis call until one of these delegated methods is called: https://github.com/chanzuckerberg/redcord/blob/master/lib/redcord/relation.rb#L69

Can we also trace the delegated methods, or is it already doing that?

Reply:
oooh, that's right! we should move this tracing to execute_query